extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientConfig.java

```diff
@@ -33,6 +33,20 @@
     @ConfigItem(defaultValue = "graphql-transport-ws")
     public Optional<List<String>> subprotocols;
 
+    /**
+     * If true, then queries and mutations will run over the websocket transport rather than pure HTTP.
+     * Off by default, because it has higher overhead.
+     */
+    @ConfigItem
+    public Optional<Boolean> executeSingleResultOperationsOverWebsocket;
```

Comment:
This can be a `boolean` because it's false by default already?

Reply:
In that case we wouldn't be able to differentiate between the cases _the user set a value that is equal to the default value_ and _the user didn't set any value, so we resort to the default_, and so wouldn't know whether the value of this property should override any configuration that comes from SmallRye (via `smallrye.graphql.*` properties or set in the programmatic builder), where the same might or might not be specified. `GraphQLClientConfigurationMergerBean` overrides the smallrye-specific configuration when relevant - it does that only if the user explicitly set a value for this. 
So we need to know whether this was explicitly specified or not.